enterprise/server/cmd/ci_runner/main.go

```diff
@@ -666,11 +666,15 @@ func parseFlags() error {
 	flagset.Init(os.Args[0], flag.ContinueOnError)
 	flagset.Usage = func() {}
 
+	isCredHelper := false
```

Comment:
I think we could get rid of the special case handling in this func and instead do something like this in the place where we call parseFlags() today

```
if slices.Contains(os.Args, "--credential_helper") {
  flag.Parse()
} else {
  // Use special flag parsing for CI runner to handle flags that aren't defined yet.
  parseFlags()
}
```

Reply:
that's a lot cleaner!

applied.

Reply:
could you also revert the `parseFlags` changes from the previous PR ? Those should no longer be needed